lib/utils/scope-util.js

```diff
@@ -3,7 +3,9 @@ const InvalidScopeError = require('../errors/invalid-scope-error');
 
 module.exports = {
   parseScope: function (requestedScope) {
-    if (!isFormat.nqschar(requestedScope)) {
+    // XXX: isFormat.nqschar will trat Arrays of strings like String,
+    // thus we additionally check, whether incoming scopes are Arrays
+    if (!isFormat.nqschar(requestedScope) || Array.isArray(requestedScope)) {
```

Comment:
I would rework this for performance reasons

```suggestion
    if (typeof requestedScope !== 'string' || !isFormat.nqschar(requestedScope)) {
```

Reply:
this, however would throw on `null` or `undefined` scope, which are supported

Reply:
good point - are those values to be treated any differently to an empty string? could we add a truthy check?

Reply:
I just played around with different values and `isFormat.nqschar` throws already on empty strings BUT it accepts Strings that contain only whitespace like `' '` which is another edge-case.

Reply:
Furthermore it also transforms `read   write` (multiple spaces) into `[ 'read', '', '', '', 'write' ]`.